packages/superset-ui-core/src/query/types/PostProcessing.ts

```diff
@@ -0,0 +1,118 @@
+import { JsonObject } from '../../connection';
+import { TimeGranularity } from '../../time-format';
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+export type NumpyFunction =
+  | 'average'
+  | 'argmin'
+  | 'argmax'
+  | 'count'
+  | 'count_nonzero'
+  | 'cumsum'
+  | 'cumprod'
+  | 'max'
+  | 'mean'
+  | 'median'
+  | 'nansum'
+  | 'nanmin'
+  | 'nanmax'
+  | 'nanmean'
+  | 'nanmedian'
+  | 'nanpercentile'
+  | 'min'
+  | 'percentile'
+  | 'prod'
+  | 'product'
+  | 'std'
+  | 'sum'
+  | 'var';
+
+interface Aggregates {
+  /**
+   * The name of the generated aggregate column.
+   */
+  [colname: string]: {
+    operator: NumpyFunction;
+    /**
+     * the name of the column to generate aggrates from.
+     */
+    column?: string;
+    options?: JsonObject;
+  };
+}
+
+export interface PostProcessingAggregation {
+  operation: 'aggregation';
```

Comment:
should it be some TS type?

Reply:
I think it's find to use string literals here, otherwise we'd need to create a type for each operation, which doesn't seem to add much additional value.